### ISSUE TITLE
Row2cell function creates vectorized cells that are off-centre from the actual raster

### ISSUE THREAD

**Issue by tastatham** [gdsbook/book#180]
The row2cell function from https://geographicdata.science/book/notebooks/03_spatial_data.html#one-pixel-at-a-time creates vectorized cells that are off centre from the actual raster. I'm not sure whether this was intentional?

I have modified the row2cell function to return (all) vectorized cells as a GeoDataFrame with accompanying pixel value for my own work - which might be useful?

```py
import pandas
import geopandas
import rioxarray
from shapely.geometry import box

def _polygonize(rioxarray_obj):
        
    poly_ls = []
    
    #crs = clipped.rio.crs #TO DO - get crs from clipped.spatial_ref 
    # Affine transformation
    af = rioxarray_obj.rio.transform()
    # Return non-negative x and y resolution
    xres, yres = abs(af[0]), abs(af[4])
    # Convert raster to to Pandas Series
    series = rioxarray_obj.to_series().reset_index(drop=False).rename(columns={0: "Value"})
    
    # Loop through each row and create grid cell
    for row in range(len(series)):
        row = series.loc[row, :]
        # Define poly bounds
        xmin = row["x"] - (xres / 2)
        xmax = row["x"] + (xres / 2)
        ymin = row["y"] + (yres / 2) 
        ymax = row["y"] - (yres / 2)
        # Create cell
        cell = box(xmin, ymin, xmax, ymax)
        poly_ls.append(cell)
           
    # Create GeoDataFrame
    gdf = pd.concat([series["Value"], geopandas.GeoSeries(poly_ls).to_frame("geometry")], axis=1)
    gdf.crs  = 4326 #TO DO - use raster crs 
    return gdf
```

**Comment by darribas** [comment]
This is fantastic @tastatham, excellent catch and thank you very much for reporting! I've taken your idea and implemented in #261, the merge should be on the way!

**Comment by darribas** [comment]
Fixed in #261
